usr/share/rear/layout/prepare/GNU/Linux/100_include_partition_code.sh

```diff
@@ -30,15 +30,6 @@ create_disk() {
     [[ -b "$disk" ]]
     BugIfError "Disk $disk is not a block device."
```

Comment:
could you (while you are at it) by the way replace all *IfError function calls
by plain normal bash code, see my comment at the *IfError functions about
"Using the ...IfError functions can result unexpected behaviour in certain cases"
in usr/share/rear/lib/_input-output-functions.sh that is currently at
https://github.com/rear/rear/blob/master/usr/share/rear/lib/_input-output-functions.sh#L773

Reply:
@jsmeix 
```
call e.g. LogPrintError()
to show an error message to the user and then exit the
diskrestore.sh script normally with non zero exit code
```
I think it is worth to have a helper function for this. Now, the question is whether to introduce new function(s), called maybe `SubshellError()` and `SubshellBugError()` and call them from the layout script, or whether make `Error()` and `BugError()` smarter by teaching them to DTRT when called from inside the layout script (this could be achieved by setting a dedicated variable at the start of the layout script, in `540_generate_device_code.sh`, and then examining it in `Error()` and `BugError()`). The advantage of the latter approach would be that in the former approach, you need to classify functions into two categories: those called from the layout (or similar) script and those called from the regular ReaR code, use `SubshellError()`/`SubshellBugError()` in the former and `Error()`/`BugError()` in the latter and never mix them together.
(The same problem arises even without introducing helpers like `SubshellError()`/`SubshellBugError()`. Even if performing what you propose manually (call e.g. LogPrintError() to show an error message to the user and then exit the diskrestore.sh script normally with non zero exit code) you still need to classify functions according to whether one should do this or one is allowed to use `Error()`/`BugError()` normally.)

Reply:
```
I am even thinking about no longer exit the whole ReaR
during "rear recover" by default when Error() is called
because this problem here (Error exit during diskrestore.sh)
shows that likely it is basically always wrong to
exit the whole ReaR during "rear recover" because
that results basically always bad UX for the user
```
Your reasoning may apply to `Error()` but not to `BugError()`. There is not much that can be done when a code bug is detected - all bets are off in this situation and nasal daemons may fly, as you can never be certain what the bug is and how to handle it. After all, calling `assert()` or a similar function that aborts in places that are considered unreachable is a common practice, and a good one, as it permits detecting problems early instead of continuing with bad data. So I think that some variant of `BugError()` that aborts ReaR will always have its uses.
Of course, this does not mean that in the layout script `BugError()` needs to abort the whole ReaR. It is reasonable to abort only the script and let the user fix it (the script is meant to be adjustable by the user, after all).

Reply:
@jsmeix 
> I leave it to you if you prefer to keep the current BugError
> in this pull request here or to change it into a LogPrintError.

I prefer to keep it as it is and enhance `Error()` (or `BugError()`) separately to behave properly in scripts executed in subshells.